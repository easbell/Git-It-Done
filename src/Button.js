import React, { Component } from 'react'

export class Button extends Component {
  
  
  sendAnswer = () => {
    const {checkAnswer} = this.props
    checkAnswer(this.props.answer)
  }

  render() {
    return (
      <div>
        <button onClick={this.sendAnswer}>{this.props.answer}</button>
      </div>
    )
  }
}

export default Button
