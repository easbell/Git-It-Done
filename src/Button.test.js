import React from 'react';
import Button from './Button';
import { shallow } from 'enzyme';

const checkAnswerMock = jest.fn();
const checkIndexMock = jest.fn();

const mockChoice = "cdir";

describe('Button', () => {
  const wrapper = shallow(
    <Button 
      answer={mockChoice}
      checkAnswer={checkAnswerMock}
      checkIndex={checkIndexMock}
    />
  );

  it('should match snapshot with all props passed in', () => {
      expect(wrapper).toMatchSnapshot();
  });
})