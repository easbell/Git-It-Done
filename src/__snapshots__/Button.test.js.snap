// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match snapshot with all props passed in 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    cdir
  </button>
</div>
`;
